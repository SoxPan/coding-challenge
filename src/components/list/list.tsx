import React, { FC } from "react";
import styled from "styled-components";

import { ListItem } from "../listItem/listItem";

import { IJob } from "../../types";

const StyledList = styled.ul`
    width: 100%;
    height: auto;
    list-style: none;
    margin: 0;
    padding: 15px;
    background: #eeeeee;
`;

interface IProps {
    jobs: IJob[];
}

export const List: FC<IProps> = ({ jobs }) => {
    return (
        <StyledList>
            {jobs.map((job: IJob) => (
                <ListItem key={job.id} job={job} />
            ))}
        </StyledList>
    );
};
