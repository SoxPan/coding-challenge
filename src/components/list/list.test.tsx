import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { List } from "../list/list";
import data from "../../../public/mock.json";

describe("<List />", () => {
    it("List renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(<List jobs={data} />, div);
    });

    it("List renders correctly", () => {
        const tree = renderer.create(<List jobs={data} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
