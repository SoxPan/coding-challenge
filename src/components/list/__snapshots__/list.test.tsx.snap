// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> List renders correctly 1`] = `
<ul
  className="sc-htpNat dhWqyy"
>
  <li
    className="sc-bdVaJa ekTLhB"
  >
    <a
      className="sc-bwzfXH hCIMcE"
      href="/1"
      onClick={[Function]}
    >
      Frontend Developer
    </a>
  </li>
  <li
    className="sc-bdVaJa ekTLhB"
  >
    <a
      className="sc-bwzfXH hCIMcE"
      href="/2"
      onClick={[Function]}
    >
      PHP Developer
    </a>
  </li>
  <li
    className="sc-bdVaJa ekTLhB"
  >
    <a
      className="sc-bwzfXH hCIMcE"
      href="/3"
      onClick={[Function]}
    >
      (Senior) Key Account Manager (m/w/d) B2B
    </a>
  </li>
</ul>
`;
